### PR TITLE
Fix null SSL* dereference in TLSSocket.getServername after close

### PR DESCRIPTION
**Pull request by robobun**
## Repro

```js
const client = await Bun.connect({
  hostname, port, tls: { ... },
  socket: {
    handshake(s) { s.end(); },
    close(s) { s.getServername(); }, // SEGV
  },
});
```

`onClose` detaches the socket (`this.socket.detach()`) before invoking the JS `close` handler, so `this.socket.ssl()` returns `null` there. `getServername` passed that null straight to `SSL_get_servername`, which dereferences `ssl->hostname` → segfault at offset 0x90.

## Cause

Every other TLS getter in `tls_socket_functions.zig` (`getCipher`, `getTLSVersion`, `getCertificate`, `getSession`, `getTLSTicket`, …) guards with `this.socket.ssl() orelse return .js_undefined`. `getServername` was the one that didn't.

## Fix

```zig
const ssl_ptr = this.socket.ssl() orelse return .js_undefined;
```

## Verification

Without fix (`bun bd`, ASAN):
```
==4116==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000090
    #0 ... std::__uniq_ptr_impl<char, bssl::internal::Deleter>::_M_ptr() const
```

With fix: test passes, `getServername()` returns `undefined` after close.

### CODE REVIEW

File: test/js/bun/net/socket.test.ts [hunk 1]
```diff
@@ -832,6 +832,45 @@ it("reading fd of a TLS listener should not crash", () => {
...
+      close(socket) {
+        // The underlying SSL* is already gone by the time close fires;
+        // getServername must return undefined rather than deref a null SSL*.
+        try {
+          resolve(socket.getServername());
+        } catch (e) {
+          reject(e);
+        }
+      },
+      error() {},
```
[the remark this answers is not in the document]
Comment (robobun) [reply test/js/bun/net/socket.test.ts]: Done in 28522bd — `error` handler now rejects the promise.
